src/crab/ebpf_domain.cpp

```diff
@@ -2154,8 +2154,61 @@ void ebpf_domain_t::do_mem_store(const Mem& b, Type val_type, SValue val_svalue,
     });
 }
 
-void ebpf_domain_t::operator()(const LockAdd& a) {
-    // nothing to do here
+void ebpf_domain_t::operator()(const Atomic& a) {
+    if (m_inv.is_bottom())
+        return;
+    if (!m_inv.entail(type_is_pointer(reg_pack(a.access.basereg))) ||
+        !m_inv.entail(type_is_number(reg_pack(a.valreg)))) {
+        return;
+    }
+
+    // Fetch the current value into the R11 pseudo-register.
+    Reg r11{R11_ATOMIC_SCRATCH};
+    Mem mem = {.access = a.access, .value = r11, .is_load = true};
+    (*this)(mem);
+
+    // Compute the new value in R11.
+    Bin bin {.dst = r11, .v = a.valreg, .is64 = (a.access.width == sizeof(uint64_t)), .lddw = false};
+    switch ((Atomic::Op)((uint32_t)a.op & (uint32_t)Atomic::Op::BASE_MASK)) {
+    case Atomic::Op::ADD: bin.op = Bin::Op::ADD; break;
+    case Atomic::Op::OR: bin.op = Bin::Op::OR; break;
+    case Atomic::Op::AND: bin.op = Bin::Op::AND; break;
+    case Atomic::Op::XOR: bin.op = Bin::Op::XOR; break;
+    case Atomic::Op::XCHG_BASE:
+    case Atomic::Op::CMPXCHG_BASE: bin.op = Bin::Op::MOV; break;
```

Comment:
How about adding XCHG nd CMPXCHG to Bin::Op and use it instead? It'll appear on some additional switches, but it will just fallthrough to MOV, unless there's actual difference.

Reply:
If not,  please move this to a function, something like `(*this)(atomic_to_bin(a))`.

Reply:
Moved to a function atomic_to_bin() since the alternative seems more bug prone to me as new instructions might get added in the future.